boxoffice/models/line_item.py

```diff
@@ -1,47 +1,212 @@
-            'category_id': self.category_id,
-            'category_desc': self.category.title,
-            'quantity': self.quantity,
-            'pat': self.category.price_after_tax,
-            'tax_rate': self.category.tax_rate,
-            'created_at': self.created_at.isoformat() + 'Z',
-            'updated_at': self.updated_at.isoformat() + 'Z',
-        }
+    __tablename__ = 'line_item'
+    __uuid_primary_key__ = True
+    __tableargs__ = (db.UniqueConstraint('order_id', 'line_item_no'),)
+
+    customer_order_id = db.Column(None, db.ForeignKey('customer_order.id'), nullable=False)
+    order = db.relationship(Order, backref=db.backref('line_items', cascade='all, delete-orphan'))
+    # line_item_no is the relative number of the line item per order.
+    line_item_no = db.Column(db.Integer, nullable=False)
+
+    item_id = db.Column(None, db.ForeignKey('item.id'), nullable=False)
+    item = db.relationship(Item, backref=db.backref('line_items', cascade='all, delete-orphan'))
+
+    discount_policy_id = db.Column(None, db.ForeignKey('discount_policy.id'), nullable=True)
+    discount_policy = db.relationship('DiscountPolicy', backref=db.backref('line_items'))
+
+    discount_coupon_id = db.Column(None, db.ForeignKey('discount_coupon.id'), nullable=True)
+    discount_coupon = db.relationship('DiscountCoupon', backref=db.backref('line_items'))
+
+    base_amount = db.Column(db.Numeric, default=decimal.Decimal(0), nullable=False)
+    discounted_amount = db.Column(db.Numeric, default=decimal.Decimal(0), nullable=False)
+    final_amount = db.Column(db.Numeric, default=decimal.Decimal(0), nullable=False)
+    status = db.Column(db.Integer, default=LINE_ITEM_STATUS.CONFIRMED, nullable=False)
+    ordered_at = db.Column(db.DateTime, nullable=True)
+    cancelled_at = db.Column(db.DateTime, nullable=True)
+
+    @classmethod
+    def make_ntuple(cls, item_id, base_amount, **kwargs):
+        line_item_tup = namedtuple('LineItem', ['item_id', 'base_amount', 'discount_policy_id', 'discount_coupon_id', 'discounted_amount'])
+        return line_item_tup(item_id,
+            base_amount,
+            kwargs.get('discount_policy_id', None),
+            kwargs.get('discount_coupon_id', None),
+            kwargs.get('discounted_amount', decimal.Decimal(0)))
+
+    @classmethod
+    def calculate(cls, line_item_dicts, coupons=[]):
+        """
+        Returns line item tuples with the respective base_amount, discounted_amount,
+        final_amount, discount_policy and discount coupon populated
+        """
+        item_line_items = dict()
+        line_items = []
+        for line_item_dict in line_item_dicts:
+            item = Item.query.get(line_item_dict.get('item_id'))
+            if not item_line_items.get(unicode(item.id)):
+                item_line_items[unicode(item.id)] = []
+            item_line_items[unicode(item.id)].append(LineItem.make_ntuple(item_id=item.id,
+                base_amount=Price.current(item).amount))
```

Comment:
`Price.current` is the wrong place to host this method. It should be `item.current_price`. You can place the code alongside the `Price` class and monkey-patch it into the `Item` class.